### ISSUE TITLE
why is is-valid added to form fields by default?

### ISSUE THREAD

**Issue by timrichardson** [zostera/django-bootstrap4#276]
I change to bootstrap4 from bootstrap3. I have a filter form on top of a table, rendered with 

```
{% if filter %}
    <form action="" method="get" class="form form-inline">
        {% bootstrap_form filter.form layout='inline' %}
        {% bootstrap_button 'filter' %}
    </form>
{% endif %}
```

The field is rendered  as 

`<input .... class="form-control is-valid"...>`

the `is-valid ` class means my field gets a big green tick even on the first page load (before submit)
Bootstrap3 didn't do this decoration.
I realise I can prevent `is-valid` via settings for the BOOTSTRAP4 app, but why is this the default?

![image](https://user-images.githubusercontent.com/927458/115137863-e2167700-a06b-11eb-932c-92595501eb9c.png)

**Comment by dyve** [comment]
Yes, for GET forms this is very illogical, I agree. Use the workaround for now, I'm working on the similar problem in `django-bootstrap5`, and I'll see if I can backport it once it's fixed there.

**Comment by timrichardson** [comment]
Thanks for the fast update. Can you update this issue if you fix the bootstrap5 version, in case you don't have time to backport it, I will try a PR.

**Comment by dyve** [comment]
Looking at https://getbootstrap.com/docs/4.6/components/forms/#validation, it seems Bootstrap 4 advises against server side validation. A solution would be to default to `False` in the settings, have `True` lead to the default Bootstrap class(es), and allow a string to customize the valid/invalid classes.

**Comment by timrichardson** [comment]
Note that I got a stackoverflow answer https://stackoverflow.com/a/67145950/401226 which suggests:
 A workaround less dramatic than changing global defaults is:
" you can pass the class that should be rendered instead as the keyword argument bound_css_class to the bootstrap_form template tag:

{% bootstrap_form filter.form layout='inline' bound_css_class='' %}
"
which is not well documented (or documented at all).


However, this doesn't address the question of a sensible default.
It seems to me that if you're using Django you are likely to use server-side validation. 
But why has the behaviour changed between bootstrap3 and 4?

**Comment by dyve** [comment]
I may have to recheck, but I think this behavior was (at least originally) the same in django-bootstrap3. Bootstrap 4 does have a much more visible "Validated" look (Bootstrap 3 just does a green line around the field).

**Comment by dyve** [comment]
Keeping the default, documenting it properly.

You are right that it is illogical for a GET filter form. django-bootstrap5 resolved this by defaulting `success_css_class` to `''`. Doing the same here would change the rendered HTML of every bound form in every project using this package, and django-bootstrap4 is in maintenance mode, so that change is out of scope.

What I have done instead is document the behaviour and both escape hatches, which were effectively undocumented. Per tag:

```
{% bootstrap_form filter.form layout='inline' bound_css_class='' %}
```

Project wide:

```python
BOOTSTRAP4 = {'success_css_class': ''}
```

Checking this also turned up that the `bootstrap_field` docs still listed the Bootstrap 3 defaults, `has-error` and `has-success`, rather than the real `is-invalid` and `is-valid`. Both are corrected in #893, along with a settings section explaining why a bound form shows a green tick on first load.

To answer your question directly: the behaviour did not change between django-bootstrap3 and 4. Bootstrap 4 just renders the valid state far more loudly, a green tick and border instead of a thin coloured outline, so the same markup became impossible to ignore.

For new projects, django-bootstrap5 has the default you are asking for.

Closing.
